x-pack/legacy/plugins/maps/public/layers/sources/source_registry.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import { AbstractSourceDescriptor } from '../../../common/descriptor_types';
+import { ISource } from './source';
+
+type SourceRegistryEntry = {
+  factory: (sourceDescriptor: AbstractSourceDescriptor, inspectorAdapters: object) => ISource;
+  type: string;
+};
+
+const registry: SourceRegistryEntry[] = [];
+
+export function registerSource(entry: SourceRegistryEntry) {
+  registry.push(entry);
+}
+
+function getSourceByType(sourceType: string): SourceRegistryEntry | undefined {
+  return registry.find((source: SourceRegistryEntry) => source.type === sourceType);
+}
+
+export function createSourceInstance(
+  sourceDescriptor: AbstractSourceDescriptor,
+  inspectorAdapters: object
+) {
```

Comment:
I'd remove the `inspectorAdapters` from the signature, since it's a constant param, and move it to source_registry

Reply:
I'd also add the return-type to the signature `export function createSourceInstance(...) : ISource`

Reply:
> I'd remove the inspectorAdapters from the signature, since it's a constant param, and move it to source_registry

Can you explain this a little more? How would it get passed to the source instance? `inspectorAdapters` comes from the redux store and the only way to get redux store state into the POJOs is via constructor args. I am not sure what you mean with this feedback.

Reply:
I thought we could just move the `getInspecrtorAdapters` call to the source_registry.ts module. Then we can keep the `createSourceInstance` method more simple, to just requiring the single source-descriptor argument (which is the only argument that varies).

but maybe that's not possible or would create a circular dependency?

Reply:
source_registry is not part of redux connected components so you can not use `getInspecrtorAdapters` from redux

Reply:
as discussed, moving `createSourceInstance` back to map_selectors but leaving inspectorAdapters in the constructor. Moving inspectorAdapters from source constructor is outside the scope of this PR.